validator_client/src/doppelganger_service.rs

```diff
@@ -182,7 +182,7 @@ async fn beacon_node_liveness<'a, T: 'static + SlotClock, E: EthSpec>(
                 OfflineOnFailure::Yes,
                 |beacon_node| async move {
                     beacon_node
-                        .post_lighthouse_liveness(validator_indices, previous_epoch)
+                        .post_validator_liveness_epoch(previous_epoch, validator_indices.to_vec())
                         .await
                         .map_err(|e| format!("Failed query for validator liveness: {:?}", e))
                         .map(|result| result.data)
```

Comment:
It would potentially be more efficient and concise if we transform the `Vec` in place rather than adding an intermediate variable to hold the untransformed data.  Also using `into_iter` in this scenario is preferred given that we no longer need the original vector and can avoid unnecessary allocations and copies.

e.g. 

```rust
beacon_node
    .post_validator_liveness_epoch(previous_epoch, validator_indices.to_vec())
    .await
    .map_err(|e| format!("Failed query for validator liveness: {:?}", e))
    .map(|result| {
        result
            .data
            .into_iter()
            .map(|response| LivenessResponseData {
                index: response.index,
                epoch: previous_epoch,
                is_live: response.is_live,
            })
            .collect()
    })
```